Shared/Samples/Edit with branch versioning/EditWithBranchVersioningView.swift

```diff
@@ -0,0 +1,250 @@
+    @State private var isCreatingVersion = false
+    
+    /// A Boolean value indicating whether the switch version dialog is presented.
+    @State private var isSwitchingVersion = false
+    
+    /// A Boolean value indicating whether the set damage type dialog is presented.
+    @State private var isSettingDamageType = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(map: model.map)
+                .callout(placement: $calloutPlacement.animation(.default.speed(2))) { placement in
+                    let placeName = placement.geoElement?.attributes[.placeNameKey] as? String
+                    let damageType = placement.geoElement?.attributes[.damageTypeKey] as? String
+                    
+                    HStack {
+                        VStack(alignment: .leading) {
+                            Text(placeName ?? "")
+                                .font(.headline)
+                            Text(damageType ?? "")
+                                .font(.callout)
+                        }
+                        editDamageTypeButton
+                    }
+                    .padding(5)
+                }
+                .onSingleTapGesture { screenPoint, mapPoint in
+                    if model.selectedFeature != nil && !model.onDefaultVersion {
+                        // Shows the move confirmation alert if there is already a feature selected.
+                        isMovingFeature = true
+                        moveLocation = mapPoint
+                    } else {
+                        // Identifies and selects the feature at the tap location.
+                        selectedAction = .selectFeature(screenPoint: screenPoint, mapPoint: mapPoint)
+                    }
+                }
+                .alert(
+                    "Confirm Move",
+                    isPresented: $isMovingFeature,
+                    presenting: moveLocation,
+                    actions: { mapPoint in
+                        Button("Cancel", role: .cancel, action: {})
+                        Button("Move") {
+                            model.selectedFeature?.geometry = mapPoint
+                            selectedAction = .updateFeature
+                        }
+                    },
+                    message: { _ in
+                        Text("Do you want to move the selected feature?")
+                    }
+                )
+                .task(id: selectedAction) {
+                    guard let selectedAction else { return }
+                    defer { self.selectedAction = nil }
+                    calloutPlacement = nil
+                    
+                    do {
+                        switch selectedAction {
+                        case .setUp:
+                            try await model.setUp()
+                            
```

Comment:
We usually don't put a blank line in between cases. Is there a particular reason why you did it here? if not, please remove them.